controllers/UserController.js

```diff
@@ -0,0 +1,93 @@
+
+const knex = require('knex')({
+    client: 'pg',
+    connection: {
+        host: 'localhost',
+        user: 'admin',
+        password: 'admin',
+        database: 'app'
+    }
+});
+
+
+const getUsersDetails = async (req, res) => {
+
+    try {
+        const customers = await knex('customers').select('*')
+        res.json(customers)
+    } catch (error) {
+        res.json('error' + error)
+    }
+};
+const getUserDetails = async (req, res) => {
+
+    const { id: customerid } = req.params;
+    try {
+        const customers = await knex('customers').where({ customerid }).first()
+        res.json(customers)
+    } catch (error) {
+        res.json('error' + error)
+    }
+};
+
+const createUser = async (req, res) => {
+
+    try {
+        const { firstname, lastname } = req.body;
+
+        const user = await knex('customers')
+            .insert({
+                firstname, lastname
+            })
+            .returning('*')
+
+        res.json(user)
+
+    } catch (error) {
+        res.json('error' + error)
+    }
+};
+
+const updateUser = async (req, res) => {
+
+    try {
+        const { id: customerid } = req.params;
+        const { firstname, lastname } = req.body;
+
+        const user = await knex('customers')
+            .update({
+                firstname, lastname
+            })
+            .where({ customerid })
+            .returning('*')
+
+        res.json(user)
+
+    } catch (error) {
+        res.json('error' + error)
```

Comment:
You can return either error or you can send it as `{error}` so that you can get it as an object with key `error`